### PR TITLE
CHROMEOS config/lava/chromeos: Add chromeos-install-modules-template

### PR DESCRIPTION
**Pull request by mgalka**
Add LAVA job template for installing chromeos modules and a accompanying test-configs.yaml configuration.

### CODE REVIEW

**Comment by mgalka** [comment]
Tested on octopus
https://lava.collabora.co.uk/scheduler/job/6078545

**Comment by gctucker** [comment]
Please rebase

File: config/core/test-configs.yaml [hunk 1]
```diff
@@ -179,6 +179,16 @@ test_plans:
...
       flashing_ramdisk: http://storage.chromeos.kernelci.org/images/rootfs/chromeos/20220412.0/bullseye-chromeos-flash/amd64/initrd.cpio.gz
 
+  chromeos-install-modules:
+    pattern: 'chromeos/chromeos-install-modules-template.jinja2'
+    params:
+      job_timeout: '60'
+      modules_url: http://images.collabora.com/lava/chromeos/modules-4.14.243.tar.gz
+      flashing_kernel: http://storage.chromeos.kernelci.org/images/kernel/v5.17.2-x86_64+x86_chromebook/kernel/bzImage
+      flashing_modules: http://storage.chromeos.kernelci.org/images/kernel/v5.17.2-x86_64+x86_chromebook/modules.tar.xz
+      flashing_rootfs: http://storage.chromeos.kernelci.org/images/rootfs/chromeos/20220412.0/bullseye-chromeos-flash/amd64/full.rootfs.tar.xz
```
Comment (gctucker) [inline config/core/test-configs.yaml:189]: The URL should have been:
http://storage.chromeos.kernelci.org/images/rootfs/chromeos/bullseye-chromeos-flash/20220412.0/amd64/full.rootfs.tar.xz
I guess that's something to fix in a follow-up PR with an image built and uploaded to the right location.

**Comment by gctucker** [comment]
> Tested on octopus https://lava.collabora.co.uk/scheduler/job/6078545

Could you please test this one more time since it has been rebased?

**Comment by gctucker** [comment]
Adding `staging-skip` label to test and merge #1170 first.

File: config/core/test-configs.yaml [hunk 2]
```diff
@@ -174,10 +174,21 @@ test_plans:
...
       docker_image: 'kernelci/chromeos-flash'
-      flashing_kernel: https://storage.chromeos.kernelci.org/images/kernel/v5.10.112-x86-chromebook/kernel/bzImage
-      flashing_modules: https://storage.chromeos.kernelci.org/images/kernel/v5.10.112-x86-chromebook/modules.tar.xz
-      flashing_rootfs: http://storage.chromeos.kernelci.org/images/rootfs/chromeos/20220412.0/bullseye-chromeos-flash/amd64/full.rootfs.tar.xz
-      flashing_ramdisk: http://storage.chromeos.kernelci.org/images/rootfs/chromeos/20220412.0/bullseye-chromeos-flash/amd64/initrd.cpio.gz
+      chromeos_image: 'chromiumos_test_image-R94-14150.19.0-octopus.bin'
+      flashing_kernel: http://storage.chromeos.kernelci.org/images/kernel/v5.17.2-x86_64+x86_chromebook/kernel/bzImage
+      flashing_modules: http://storage.chromeos.kernelci.org/images/kernel/v5.17.2-x86_64+x86_chromebook/modules.tar.xz
+      flashing_rootfs: http://storage.chromeos.kernelci.org/images/rootfs/chromeos/20220505.0/bullseye-chromeos-flash/amd64/full.rootfs.tar.xz
+      flashing_ramdisk: http://storage.chromeos.kernelci.org/images/rootfs/chromeos/20220505.0/bullseye-chromeos-flash/amd64/initrd.cpio.gz
```
Comment (gctucker) [inline config/core/test-configs.yaml:181]: I don't think the move from 5.10 to 5.17 was necessary if 5.10 worked, or was there any issue with it?

If 5.17 works then great, but if we start seeing issues we might have to go back to 5.10 which should be more stable.

File: config/rootfs/debos/overlays/chromeos-flash/opt/chromeos/install-modules [hunk 3]
```diff
@@ -18,7 +18,7 @@ install_modules()
     cd "$mount_dir"
     ls -l lib/modules
     rm -rf lib/modules
-    tar xzf "$root_path/$modules_tarball"
+    tar xf "$root_path/$modules_tarball"
```
Comment (gctucker) [inline config/rootfs/debos/overlays/chromeos-flash/opt/chromeos/install-modules:21]: Maybe this could have been done in a separate commit

File: config/core/test-configs.yaml [hunk 4]
```diff
@@ -174,10 +174,21 @@ test_plans:
...
+
+  chromeos-install-modules:
+    pattern: 'chromeos/chromeos-install-modules-template.jinja2'
+    params:
+      job_timeout: '60'
+      modules_url: http://images.collabora.com/lava/chromeos/modules-4.14.243.tar.gz
```
Comment (gctucker) [inline config/core/test-configs.yaml:187]: The modules should be populated from the build meta-data.  We could do something special for "fixed kernels" like with QEMU but it seems this would be a follow-up. So I think we should drop this for now.

File: config/core/test-configs.yaml (hunk 1 above)
Comment (mgalka) [reply config/core/test-configs.yaml]: Exactly.
The URL is dependent on a location where the Jenkins job pushes the image. It may be worth filing an issue to fix the Jenkins job in the first place.

File: config/rootfs/debos/overlays/chromeos-flash/opt/chromeos/install-modules (hunk 3 above)
Comment (mgalka) [reply config/rootfs/debos/overlays/chromeos-flash/opt/chromeos/install-modules:21]: good point. I'll change that

File: config/core/test-configs-chromeos.yaml [hunk 5]
```diff
@@ -1,5 +1,9 @@
 test_plans:
 
+  chromeos-boot:
+    pattern: 'chromeos/chromeos-boot-chromebook-template.jinja2'
+    prompt_command: 'NETDEV_UP'
```
Comment (gctucker) [inline config/core/test-configs-chromeos.yaml:5]:
```suggestion
    params:
      prompt_command: 'sof-audio-pci'
```

File: config/lava/chromeos/chromeos-boot-chromebook-template.jinja2 [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{% extends 'base/kernel-ci-base.jinja2' %}
+{% block main %}
+{{ super () }}
+context:
+  extra_kernel_args: console_msg_format=syslog cros_secure cros_debug boot=/dev/{{ flashing_device }}p3
```
Comment (gctucker) [inline config/lava/chromeos/chromeos-boot-chromebook-template.jinja2:5]:
```suggestion
  extra_kernel_args: console_msg_format=syslog cros_secure cros_debug root=/dev/{{ flashing_device }}p3
```

**Review by gctucker — COMMENTED** [review]
A couple of typos but I'm sending a PR to fix them on top of this, so we can merge this one now and make progress.
